students/preactive/Lesson3/slice.py

```diff
@@ -0,0 +1,86 @@
+def slicer():
+    a_string = "this is a string"
+    a_tuple = (2, 54, 13, 12, 5, 32)
+    a_arr = ["!", "!", "tide", "pod", "challenge", "is",
+             "not", "for", "the", "weak", "!", "!"]
+    three_types = (a_string, a_tuple, a_arr)
+
+    for slice_item in three_types:
+        for_type = type(slice_item)
+        first_last(slice_item, for_type)
+        odd_out(slice_item, for_type)
+        ambigus_ao_fourth(slice_item, for_type)
+        reversed(slice_item, for_type)
+        thirds(slice_item, for_type)
+
+
+def first_last(slicer_input, for_type):
+    # with the first and last items exchanged
+    swapped = slicer_input[-1:] + slicer_input[1:len(slicer_input) - 1] \
+              + slicer_input[:1]
+    if for_type is str:
+        assert swapped == "ghis is a strint", "String Failed"
+    if for_type is tuple:
+        assert swapped == (32, 54, 13, 12, 5, 2), "Tuple Failed"
+    if for_type is list:
+        assert swapped == ["!", "!", "tide", "pod", "challenge", "is",
+             "not", "for", "the", "weak", "!", "!"], "List failed"
```

Comment:
For your assert question.. I would keep asserts separate from the actual code.


in real project tests would go at the root of the project in the `tests` folder, in small scripts like this they would go into `if __name__ == '__main__'` block

which would now make your test logic much simpler:

`assert first_last("This is a string") == "ghis is a strint"` without the need for type checking

Reply:
Moved asserts into name/main and refactored.